mira/modeling/__init__.py

```diff
@@ -42,9 +36,9 @@ def get_parameter_key(transition_key, action):
 UNHANDLED_TYPES = set()
 
 
-class Model:
-    def __init__(self, template_model):
-        self.template_model = template_model
+class Model(Generator):
```

Comment:
I don't really like that `Model` is derived from `Generator`. It's mainly meant to be a representation class that for convenience does generation when initialized from a `TemplateModel`. I'm not quite sure what we gain from adding this parent class, it doesn't do much anyway.